### PR TITLE
malus: dont panic on missing validation data

### PR DESCRIPTION
**Pull request by ordian**
Observed spurious failures when testing https://github.com/paritytech/polkadot/pull/6811.
Sometimes, we get `AssumptionCheckOutcome::BadRequest` from validation data request, I guess because of pruning
```
WARN tokio-runtime-worker parachain::runtime-api: cannot query the runtime API version: Api called for an unknown Block: State already discarded
```

### CODE REVIEW

File: node/malus/Cargo.toml [hunk 1]
```diff
@@ -37,6 +37,7 @@ rand = "0.8.5"
 
 [features]
 default = []
+fast-runtime = ["polkadot-cli/fast-runtime"]
```
Comment (sandreim) [inline node/malus/Cargo.toml:40]: Is this needed?

**Review by sandreim — APPROVED** [review]

File: node/malus/Cargo.toml (hunk 1 above)
Comment (ordian) [reply node/malus/Cargo.toml:40]: I'm not 100% sure anymore, but it doesn't hurt either :rofl:

**Comment by ordian** [comment]
> I find those errors a bit disturbing. Is the discarded state expected and if so why?

I encountered this only on my local potato machine running multiple nodes. But also seen this sometimes in the kusama logs `Assumption Check: Bad request` from time to time. I think this is due to forks and a race with finality/pruning.

**Comment by ordian** [comment]
bot merge

**Pull request merged**